server/service/labels.go

```diff
@@ -168,19 +168,33 @@ func (svc *Service) GetLabel(ctx context.Context, id uint) (*fleet.Label, error)
 ////////////////////////////////////////////////////////////////////////////////
 
 type listLabelsRequest struct {
-	ListOptions fleet.ListOptions `url:"list_options"`
+	LabelListOptions fleet.LabelListOptions `url:"label_options"`
 }
 
 type listLabelsResponse struct {
 	Labels []labelResponse `json:"labels"`
 	Err    error           `json:"error,omitempty"`
 }
 
+type listLabelsSummaryResponse struct {
+	Labels []*fleet.LabelSummary `json:"labels"`
+	Err    error                 `json:"error,omitempty"`
+}
+
 func (r listLabelsResponse) error() error { return r.Err }
 
 func listLabelsEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
 	req := request.(*listLabelsRequest)
-	labels, err := svc.ListLabels(ctx, req.ListOptions)
+
+	if req.LabelListOptions.Summary {
```

Comment:
It feels like it would be cleaner to have a new endpoint for the summary and leave the listLabelsEndpoint untouched. Have you considered that path?

Reply:
@chiiph Yes, although I seem to recall some prior discussion of favoring adding options to an existing endpoint rather than adding new endpoints. For me, either approach works just as well so it is more a question of which pattern we want to encourage for interface needs going forward.
cc @lukeheath

Reply:
Ok, if that's the preferred way, that's fine. I'm just thinking of simplicity both of code and of understanding the APIs.

Reply:
@gillespi314 Great point. I'm comfortable with either approach as this is an internal endpoint. In general, I favor separate endpoints with dedicated use cases vs. query params as I believe dedicated endpoints are easier to reason about. It also makes the docs simpler to read. 

Currently, the only summary option we have in the API _does_ have a dedicated endpoint at [fleet/host_summary](https://fleetdm.com/docs/using-fleet/rest-api#get-hosts-summary). @gillespi314 If you are inclined, we could change this to `GET /labels/summary`, which I think would be the better design approach.